### PR TITLE
Fix for regression in Node2Code.

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Fix for [regression ](https://autodesk.slack.com/archives/C1866AFQB/p1608302596349400)in Node2Code.

n2c

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### FYIs
@saintentropy

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@aparajit-pratap thanks for fixing this so quickly, do you think we can add a test case for `+` or a similar class of errors?

If you prefer to make this into a task let me know (or @mmisol  may have already filed one?)

File: src/DynamoCore/Engine/ShortestQualifiedNameReplacer.cs [hunk 1]
```diff
@@ -106,7 +106,7 @@ public override AssociativeNode VisitIdentifierNode(IdentifierNode node)
             // return IdentifierNode if identifier string is not separated by '.'
             if (strIdentList.Length == 1)
             {
-                return new IdentifierNode(strIdentList[0]);
+                return node;
```
Comment (mjkkirschner) [inline src/DynamoCore/Engine/ShortestQualifiedNameReplacer.cs:109]: If my memory serves correctly, then the AST walkers are a bit inconsistent when it comes to returning new AST nodes or the same node - maybe this should return a copy of the node instead?

Comment (aparajit-pratap) [reply src/DynamoCore/Engine/ShortestQualifiedNameReplacer.cs:109]: The AST walkers are meant to return new nodes _only_ if they are rewritten and return the original node if there is no change.

Comment (mjkkirschner) [reply src/DynamoCore/Engine/ShortestQualifiedNameReplacer.cs:109]: ah! Is that documented somewhere? Would it make sense to add a note about that to the base class or something? Not that it needs to be part of this pr.

Comment (aparajit-pratap) [reply src/DynamoCore/Engine/ShortestQualifiedNameReplacer.cs:109]: Can do.

**Review by mmisol — APPROVED** [review]
Looks good to me. Thanks for fixing @aparajit-pratap !

**Pull request merged**